Cblum cleanup

mulbc opened the pull request:
Not tested yet, prettified the distributed_fio server and client yaml files to make them easier to read

@dry923 - this touches mostly your code

dry923 commented:
/rerun all

dry923 commented:
@mulbc can you please rebase and resolve the conflicts? Thanks!

jtaleric commented on roles/fio_distributed/templates/servers.yaml:
```diff
@@ -59,15 +59,23 @@ spec:
-        args: ["python3 stockpile-wrapper.py -s {{ elasticsearch.server }} -p {{ elasticsearch.port }} -u {{ uuid }} -n $my_node_name -N $my_pod_name --redisip {{ bo.resources[0].status.podIP }} --redisport 6379 --force"]
-{% else %}
-        args: ["python3 stockpile-wrapper.py -s {{ elasticsearch.server }} -p {{ elasticsearch.port }} -u {{ uuid }} -n $my_node_name -N $my_pod_name --redisip {{ bo.resources[0].status.podIP }} --redisport 6379"]
+        args:
```
:) should all the stockpile runs (across benchmarks) be updated to be "pretty"

rsevilla87 replied:
fixed by https://github.com/cloud-bulldozer/ripsaw/pull/383

dry923 commented:
/rerun all

mulbc commented:
I don't see why the test failed, but with the recent Dockerfile update, we have switched to running in python3 now... which have broken a few things in the main.yml.
I have fixed these in my new PR regarding fio in VMs

dry923 commented:
/rerun all

dry923 commented:
> I don't see why the test failed, but with the recent Dockerfile update, we have switched to running in python3 now... which have broken a few things in the main.yml.
> I have fixed these in my new PR regarding fio in VMs

@mulbc Looking at the output it is error-ing when grepping in the logs. This grep has recently become redundant since we check ES for a definitive success and I've opened (https://github.com/cloud-bulldozer/ripsaw/pull/356) to remove them. That said, I don't know if that will fix the issue or just move the error elsewhere. I'll re-test this again once 356 is merged.

mulbc commented:
Yes this is still relevant and I'm waiting on the tests to pass so that @dry923 can merge it

dry923 commented:
/rerun all

mulbc commented:
@dry923 - can you make the output of the tests available to me so I can see that this gets merged?

dry923 commented:
@mulbc apologies for the delay. We had some major hardware issues and the CI env was down and needed a rebuild. I'm going to re-run this now

dry923 commented:
/rerun all

rsevilla87 commented:
/rerun all

jtaleric commented:
/rerun fiod

rsevilla87 commented:
/rerun all

jtaleric commented:
/rerun all

dry923 commented on roles/fio_distributed/templates/client.yaml:
```diff
@@ -36,25 +36,27 @@ spec:
-             cat /tmp/fio/fiojob-prefill;
-             mkdir -p /tmp/fiod-{{ uuid }}/fiojob-prefill;
-             fio --client=/tmp/host/hosts /tmp/fio/fiojob-prefill --output-format=json;
+          - cat /tmp/host/hosts;
```
@mulbc after testing this a few times this change does not work when adding each item to its own -'d line as it stands. It would only run correctly when listed out like before, in a "" list. If this is something you want to keep working on that's fine however it does not pass ci like this.

mulbc replied:
Got it - I'll look into this when I have some time and see how I can fix this....

dry923 commented on roles/fio_distributed/templates/client.yaml:
```diff
@@ -70,15 +72,23 @@ spec:
-        args: ["python3 stockpile-wrapper.py -s {{ elasticsearch.server }} -p {{ elasticsearch.port }} -u {{ uuid }} -n $my_node_name -N $my_pod_name --redisip {{ bo.resources[0].status.podIP }} --redisport 6379 --force"]
-{% else %}
-        args: ["python3 stockpile-wrapper.py -s {{ elasticsearch.server }} -p {{ elasticsearch.port }} -u {{ uuid }} -n $my_node_name -N $my_pod_name --redisip {{ bo.resources[0].status.podIP }} --redisport 6379"]
+        args:
```
@mulbc I like this layout much better than the single line approach that was in there prior. That said it runs into the same errors as above and does not actually execute as is. However, changing it to this did work:

```
        args:
          - "python3
             stockpile-wrapper.py
             -s {{ elasticsearch.server }}
             -p {{ elasticsearch.port }}
             -u {{ uuid }}
             -n $my_node_name
             -N $my_pod_name
             --redisip {{ bo.resources[0].status.podIP }}
             --redisport 6379
{% if metadata is defined and metadata.force|default(false) %}
             --force
{% endif %}
             "
```
